stackslib/src/net/connection.rs

```diff
@@ -1632,31 +1631,28 @@ mod test {
 
                             cnt
                         }
-                        Err(e) => match e {
-                            net_error::PermanentlyDrained => {
+                        Err(e) => {
+                            if let net_error::PermanentlyDrained = e {
```

Comment:
Could move the error matching up into the `match`:

```suggestion
                        Err(net_error::PermanentlyDrained) => {
```